### PR TITLE
feat: compiler register hook

### PR DESCRIPTION
**Pull request by tvanhens**
Fixes #

### CODE REVIEW

File: register.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    parser: {
+      syntax: "typescript",
+      dynamicImport: false,
+      decorators: false,
+      hidden: {
+        jest: true,
+      },
+    },
+    transform: null,
+    target: "es2021",
```
Comment (sam-goodwin) [inline register.js:14]: How did we choose this? Any reason not to jump straight to 2022?

**Review by sam-goodwin — APPROVED** [review]

File: register.js (hunk 1 above)
Comment (tvanhens) [reply register.js:14]: I cargo-culted it from the `test-app` directory like the grown up developer I am 🤣 

I'll update it to es2022

File: register.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+const register = require("@swc/register/lib/node").default;
+
+register({
```
Comment (sam-goodwin) [inline register.js:3]: Will we need to allow users to specify any of these configuration properties?

**Review by sam-goodwin — APPROVED** [review]

File: register.js (hunk 2 above)
Comment (tvanhens) [reply register.js:3]: Yes, but breaking this into a separate file and allowing the config to be overwritten seems to be causing the error I was running into yesterday. As it stands, users can still overwrite the config by including a `.swcrc`.

Comment (tvanhens) [reply register.js:3]: My inclination is to get the critical path in place first to unblock the getting started section of the website and then come back and figure out the problem with making this composable. Doing so is a non-breaking change with the hook in place so I don't think it makes sense to block finishing the getting started guide.

File: test-app/.swcrc [hunk 3]
```diff
@@ -1,29 +0,0 @@
-{
```
Comment (sam-goodwin) [inline test-app/.swcrc:1]: Did you forget to remove the root `.swcrc`?

**Review by thantos — APPROVED** [review]

File: test-app/.swcrc (hunk 3 above)
Comment (tvanhens) [reply test-app/.swcrc:1]: It's still needed for testing. Will remove when I add a jest loader.

**Pull request merged**